fdbserver/Ratekeeper.actor.cpp

```diff
@@ -840,14 +846,84 @@ ACTOR Future<Void> monitorThrottlingChanges(RatekeeperData *self) {
 	}
 }
 
+Future<Void> refreshStorageServerCommitCost(RatekeeperData *self) {
+	double elapsed = now() - self->lastBusiestCommitTagPick;
+	if(elapsed <= 0) return Void();
+	// for each SS, select the busiest commit tag from ssTagCommitCost
+	for(auto it = self->storageQueueInfo.begin(); it != self->storageQueueInfo.end(); ++it) {
+		it->value.busiestWriteTag.reset();
+		TransactionTag busiestTag;
+		TransactionCommitCostEstimation maxCost;
+		double maxRate = 0, maxBusyness = 0;
+		for(const auto& [tag, cost] : it->value.tagCostEst) {
+			// opsBeforeSample / opsSampled = COMMIT_SAMPLE_BYTE / 1
+			double rate = CLIENT_KNOBS->COMMIT_SAMPLE_BYTE * cost.getOpsSum() / elapsed;
+			if(rate > maxRate) {
+				busiestTag = tag;
+				maxRate = rate;
+				maxCost = cost;
+			}
+		}
+		if(maxRate > SERVER_KNOBS->MIN_TAG_PAGES_RATE) {
+			it->value.busiestWriteTag = busiestTag;
+			ASSERT(it->value.totalWriteBytes > 0 && it->value.totalWriteOps > 0);
+			maxBusyness = 0.5 * (maxCost.getBytesSum() / it->value.totalWriteBytes + maxCost.getOpsSum() / it->value.totalWriteOps);
```

Comment:
If a tag clears a massive amount of data with a single mutation, its `maxBusyness` will be at most `~0.5`.

I think a more accurate way to compute the cost of a set of mutations would be to compute a single value that is a weighted sum of the number of operations and number of bytes. e.g.:

```
maxBusyness = (OP_WEIGHT*maxCost.getOpsSum() + maxCost.getBytesSum()) / (OP_WEIGHT*it->value.totalWriteOps + it->value.totalWriteBytes);
```